code/game/objects/machinery/autodoc.dm

```diff
@@ -762,6 +762,16 @@
 		start_processing()
 		for(var/obj/O in src)
 			qdel(O)
+		if(automaticmode)
+			say("Automatic mode engaged, initialising procedures.")
+			addtimer(CALLBACK(src, .proc/auto_start), 5 SECONDS)
+
+///Callback to start auto mode on someone entering
+/obj/machinery/autodoc/proc/auto_start()
+	if(!occupant || !automaticmode)
```

Comment:
This's there to make sure auto hasn't been turned off in the last five seconds, in case you put someone in when you don't want it to full auto. It lets you set the autodoc back to manual during the delay period to prevent starting.